### PR TITLE
CSI-1090 handle blocking and releasing of utxos

### PR DESCRIPTION
**Pull request by HarishRocks**

### CODE REVIEW

**Comment by irshadCypherock** [comment]
Task linked: [CSI-1090 Block UTXOs after creating a transaction](https://app.clickup.com/t/37308523/CSI-1090)

File: src/bitcoinForks/index.ts [hunk 1]
```diff
@@ -745,17 +748,22 @@ export default class BitcoinWallet implements Partial<IWallet> {
...
+          address: txref.address,
+          txId: txref.txid,
+          vout: txref.vout,
+          value: parseInt(txref.value, 10),
+          block_height: txref.height,
+          confirmations: txref.confirmations
+        };
+        const transaction = await this.transactionDB?.getOne({
+          hash: txref.txid
+        });
```
Comment (irshadCypherock) [inline src/bitcoinForks/index.ts:816]: Throw error when transactionDb is not present. Check how addressDB was mandated on required functions.

**Review by irshadCypherock — CHANGES_REQUESTED** [review]

File: src/bitcoinForks/index.ts [hunk 2]
```diff
@@ -50,14 +50,16 @@ export default class BitcoinWallet implements Partial<IWallet> {
...
 
   constructor(
     xpub: string,
     coinType: string,
     walletId: string,
     zpub?: string,
-    addressDb?: AddressDB
+    addressDb?: AddressDB,
+    transactionDb?: TransactionDB
   ) {
```
Comment (irshadCypherock) [inline src/bitcoinForks/index.ts]: The params are getting large, use object parameters here `{ xpub: string .... })`

File: src/bitcoinForks/index.ts (hunk 1 above)
Comment (HarishRocks) [reply src/bitcoinForks/index.ts:816]: added the check in all relevant functions

**Review by Suraj-Tiwari — APPROVED** [review]

**Pull request merged**